projects/Amlogic/patches/linux/linux-001-PR94.patch

```diff
@@ -0,0 +1,92 @@
+From 4e1f6f11f977764a12390ad3ca88abfe75d2b1b4 Mon Sep 17 00:00:00 2001
+From: kszaq <kszaquitto@gmail.com>
+Date: Thu, 1 Feb 2018 22:00:38 +0100
+Subject: [PATCH 1/2] drivers/amlogic/display/osd: fix getting parameters from
+ dt
+
+If property is not found, we should not set parameter according to prop_idx as it may have been got from previous dt parameter.
+---
+ drivers/amlogic/display/osd/osd_fb.c | 15 +++++++++------
+ 1 file changed, 9 insertions(+), 6 deletions(-)
+
+diff --git a/drivers/amlogic/display/osd/osd_fb.c b/drivers/amlogic/display/osd/osd_fb.c
+index 3c1f831ea1da..93b326badd85 100644
+--- a/drivers/amlogic/display/osd/osd_fb.c
++++ b/drivers/amlogic/display/osd/osd_fb.c
+@@ -2514,13 +2514,15 @@ static int osd_probe(struct platform_device *pdev)
+ 
+ 	/* get meson-fb resource from dt */
+ 	prop = of_get_property(pdev->dev.of_node, "scale_mode", NULL);
+-	if (prop)
++	if (prop) {
+ 		prop_idx = of_read_ulong(prop, 1);
+-	osd_set_free_scale_mode_hw(DEV_OSD0, prop_idx);
++		osd_set_free_scale_mode_hw(DEV_OSD0, prop_idx);
++	}
+ 	prop = of_get_property(pdev->dev.of_node, "4k2k_fb", NULL);
+-	if (prop)
++	if (prop) {
+ 		prop_idx = of_read_ulong(prop, 1);
+-	osd_set_4k2k_fb_mode_hw(prop_idx);
++		osd_set_4k2k_fb_mode_hw(prop_idx);
++	}
+ 	/* get default display mode from dt */
+ 	ret = of_property_read_string(pdev->dev.of_node,
+ 		"display_mode_default", &str);
+@@ -2529,9 +2531,10 @@ static int osd_probe(struct platform_device *pdev)
+ 	else
+ 		current_mode = vmode_name_to_mode(str);
+ 	prop = of_get_property(pdev->dev.of_node, "pxp_mode", NULL);
+-	if (prop)
++	if (prop) {
+ 		prop_idx = of_read_ulong(prop, 1);
+-	osd_set_pxp_mode(prop_idx);
++		osd_set_pxp_mode(prop_idx);
```

Comment:
Shouldn't the changes in this file be fixed in the kernel source rather than just patching it?

Reply:
Yes I agree this should go to the kernel

Reply:
Ok, I'll push the fix for it when I get off work.

Reply:
Keep in mind that if you apply the second patch, you need to update mali libs or you will have vsync not working.

Reply:
Ok, but the change above looks like a bug in the kernel that should be fixed, regardless of whether we take this PR or not.